dao-modules/src/governance_voting.rs

```diff
@@ -0,0 +1,330 @@
+    Address, Error, Mapping, Variable,
+};
+use casper_types::{runtime_args, RuntimeArgs, U256};
+
+use self::{
+    events::{
+        FormalVotingEnded, InformalVotingEnded, VoteCast, VotingContractCreated, VotingCreated,
+    },
+    vote::Vote,
+    voting::{Voting, VotingId},
+};
+
+/// The Governance Voting module.
+
+#[derive(Instance)]
+pub struct GovernanceVoting {
+    pub variable_repo: Variable<Option<Address>>,
+    pub reputation_token: Variable<Option<Address>>,
+    pub votings: Mapping<U256, Voting>,
+    pub votes: Mapping<(U256, Address), Vote>,
+    pub voters: Mapping<U256, Vec<Option<Address>>>,
+    pub votings_count: Variable<U256>,
+    pub dust_amount: Variable<U256>,
+}
+
+impl GovernanceVoting {
+    /// Initialize the module.
+    pub fn init(&mut self, variable_repo: Address, reputation_token: Address) {
+        self.variable_repo.set(Some(variable_repo));
+        self.reputation_token.set(Some(reputation_token));
+        // Emit the event
+        let event = VotingContractCreated {
+            variable_repo,
+            reputation_token,
+            repo_voter: self_address(),
+        };
+
+        emit(event);
+    }
+
+    pub fn create_voting(&mut self, voting: &Voting, stake: U256) {
+        // Add Voting
+        self.votings.set(&voting.voting_id, voting.clone());
+        self.votings_count.set(voting.voting_id + 1);
+
+        // Emit the event
+        let event = VotingCreated {
+            creator: caller(),
+            voting_id: voting.voting_id,
+            stake,
+        };
+
+        emit(event);
+
+        // Cast first vote in favor
+        self.vote(voting.voting_id, true, stake);
+    }
+
+    pub fn finish_voting(&mut self, voting_id: VotingId) {
+        let voting = self.votings.get(&voting_id);
+
+        if voting.completed {
+            revert(Error::FinishingCompletedVotingNotAllowed)
+        }
+
+        match voting.get_voting_type() {
+            voting::VotingType::Informal => self.finish_informal_voting(voting),
+            voting::VotingType::Formal => self.finish_formal_voting(voting),
+            voting::VotingType::Unknown => revert(Error::MalformedVoting),
+        }
+    }
+
+    fn finish_informal_voting(&mut self, mut voting: Voting) {
+        if !voting.is_in_time(get_block_time()) {
+            revert(Error::InformalVotingTimeNotReached)
+        }
+
+        let voters = self.voters.get(&voting.voting_id);
+
+        let result: Option<&str> = match voting.get_result(voters.len()) {
+            voting::VotingResult::InFavor => {
+                self.return_reputation(voting.voting_id);
+
+                // Formal voting is created with an initial stake
+                let formal_voting =
+                    voting.convert_to_formal(self.votings_count.get(), get_block_time());
+                let creator_address = voters.first().unwrap().unwrap();
+                self.create_voting(
```

Comment:
If Alice creates a informal vote and then Bob calls `finish_informal_vote`, then call to `create_voting` would emit event `VotingCreated` with Bob as creator, but it should be Alice.

Reply:
You're right, fixed